src/data/polynomial/ring_division.lean

```diff
@@ -385,6 +385,10 @@ calc ((roots ((X : polynomial R) ^ n - C a)).card : with_bot ℕ)
       ≤ degree ((X : polynomial R) ^ n - C a) : card_roots (X_pow_sub_C_ne_zero hn a)
   ... = n : degree_X_pow_sub_C hn a
 
+lemma polynomial.eval_zero_eq_roots {f : polynomial R} (hf : f ≠ 0) :
+  (λ (a : R), polynomial.eval a f = 0) = (λ (a : R), a ∈ f.roots.to_finset) :=
```

Comment:
It seems odd to me to state this as an equality between lambdas. Is there a reason for not using the goal state after `ext` as the statement?

Reply:
Your suggestion works, thanks!  This was a technical lemma, used only in a ```rw```.  The ```rw``` stays the same with the reformulation, so I am happy with the change!

In any case, I am moving this lemma and a couple of extra lemmas in a separate file.